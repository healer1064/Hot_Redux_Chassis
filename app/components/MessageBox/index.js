// @flow

import styles from './MessageBox.css';
import React from 'react';
import cx from 'classnames';
import Icon from '../Icon';

type Props = {
  message: string | React.Element<*>,
  type: 'success' | 'warning' | 'error',
  onClose?: () => void
};

function MessageBox({ message, type, onClose }: Props) {
  return (
    <div className={cx(styles.messageBox, styles[type])}>
      {message}
      <button className={styles.button} onClick={onClose}>
        <Icon name="close" />
      </button>
    </div>
  );
}

export default MessageBox;
