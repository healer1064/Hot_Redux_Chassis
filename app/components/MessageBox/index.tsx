import styles from './MessageBox.css';
import React, { ReactNode } from 'react';
import cx from 'classnames';
import Icon from '../Icon';

type Props = {
  message: ReactNode;
  type: 'success' | 'warning' | 'error' | 'neutral';
  onClose?: () => void;
};

function MessageBox({ message, type, onClose }: Props) {
  return (
    <div className={cx(styles.messageBox, styles[type])}>
      {message}
      {onClose && (
        <button className={styles.button} onClick={onClose}>
          <Icon name="close" />
        </button>
      )}
    </div>
  );
}

export default MessageBox;
