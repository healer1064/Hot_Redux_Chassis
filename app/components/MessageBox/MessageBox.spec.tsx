import MessageBox from './';
import { testComponentSnapshots } from 'tests/helpers';

const defaultProps = {
  message: 'Hello World',
  type: 'success' as 'success'
};

describe('<MessageBox />', () => {
  testComponentSnapshots(MessageBox, [defaultProps]);
});
