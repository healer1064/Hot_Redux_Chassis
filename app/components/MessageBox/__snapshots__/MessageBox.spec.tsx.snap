// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageBox /> should match snapshot 1`] = `
<div
  className="messageBox success"
>
  Hello World
</div>
`;
